NewForce-Cohort-6/tabloidmvc-candy-corn
# TS || ViewAllPosts / DeletePosts Tickets 4 and 7

# Description

**This code resolves ticket #7(Delete Posts). This should allow a given user to delete their own posts.

https://trello.com/b/teDb9crL/the-candy-corns

Given an author is viewing a Post that they have written
When they select the delete option
Then they should be presented to confirm the deletion

Given the author wishes to confirm the delete
When they the select the option to confirm
Then the Post should be removed from the system
And the author should be directed back to the Post list

Given the author decides not to confirm the delete
When they the select the option to reject confirmation
Then the Post should NOT be removed from the system
And the author should be directed back to the Post details

**This code also resolves ticket #4(View Current User's Posts)

https://trello.com/b/teDb9crL/the-candy-corns

Given the user is in the Tabloid application
When they select the My Posts menu option
Then they should be directed to the "My Posts" list page
And the page should display ALL the Posts authored by the logged-in user
And each post in the list should display the title, author and category
And the list should be in order of creation date with the most recent on top


____________________________________
# Testing Instructions for Ticket 7

* In your terminal, from your current branch:
-git add and git commit whatever you're working on in your branch
-git checkout main
-git fetch --a or git fetch all
-(if files do not show, git pull origin)
-git checkout TSdeletePost#7
-start the application with "start <application name>"

* Ensure that you have created the Tabloid database
* If needed:

-Use this to fill out tables and columns: 

[SQL data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-bit-o-honeys/blob/main/SQL/01_TabloidMVC_Create_DB.sql)

-Use this to seed the database: 

[SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)

* Launch the app from VisualStudio using the green outlined "play" button
* Ensure you have at least two users in the database by performing a sql query on the TabloidMVC UserProfile database table, if not:
 -Navigate to the login screen after running the application in the browsing
 -select the register option and create a new user
 -Navigate to the posts tab from the navbar
 -select "create new post" and create a new post
 
* Once you have created at least one post and can see at least two posts from different users in the posts tab of the app, attempt to delete the post from the account you are logged in as. 
* You should be taken to a view of that post with a confirmation to Delete.
* Confirm the delete by hitting the delete button again. This should remove the post from the database and return you to the list of all posts.
* The post that was deleted should not appear in the list of all posts.
*  You can double check by performing a sql query on the Post database table

* Attempt to delete another user's post.
* You should be shown a message that says you are not authorized with an option to return to the list

_____________________________________
# Testing Instructions for Ticket 4

* In your terminal, from your current branch:
-git add and git commit whatever you're working on in your branch
-git checkout main
-git fetch --a or git fetch all
-(if files do not show, git pull origin)
-git checkout TSdeletePost#7
-start the application with "start <application name>"

* Ensure that you have created the Tabloid database
* If needed:

-Use this to fill out tables and columns: 

[SQL data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-bit-o-honeys/blob/main/SQL/01_TabloidMVC_Create_DB.sql)

-Use this to seed the database: 

[SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)

* Launch the app from VisualStudio using the green outlined "play" button
* If not logged in, log in by selecting the login button and submitting your email
* Once on the homepage, select the "My Posts" tab from the navbar
* Once selected, you see 



____________________
# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ x] My changes generate no new warnings or errors
- [ x] I have added test instructions that prove my fix is effective or that my feature works
